### PR TITLE
Spider for broken links during npm test.

### LINKED ISSUE #804: Add a test that checks for broken links
It shouldn't be too hard to use a library like [simplecrawler](https://www.npmjs.com/package/simplecrawler) to crawl the site and ensure that, at the very least, all internal links (i.e. links that don't go to external sites) are valid. This could prevent issues like #603 and #714, and if we crawl `srcset` attributes it can also prevent ones like #664.

Before deploying to production, we could also automatically check to ensure that no links are going to `fixme` (#500, #597).

(Originally this issue also included searching for huge images, but that has been moved to #810).

### PR DESCRIPTION
**Pull request by toolness**
This fixes #804. Example output on error:

On success:

### CODE REVIEW

**Comment by toolness** [comment]
Travis is going to fail because the spidering actually finds broken links, causing `npm test` to fail. :)

Specifically, it's running into #714.

**Comment by toolness** [comment]
Unfortunately, the spidering library we're using doesn't seem to parse `srcset` tags, so not all of our responsive images will be checked, which is a bummer. 

Still, this is better than nothing right now.

**Comment by toolness** [comment]
Ok, this now fixes #714 using the suggestion described in https://github.com/mozilla/teach.webmaker.org/issues/714#issuecomment-95921298, since the spidering tests can't actually pass unless there are no 404s.

**Comment by toolness** [comment]
@ScottDowne can you take a look at this?

File: less/index.less [hunk 1]
```diff
@@ -4,6 +4,3 @@
 @import "components/index";
 @import "pages/index";
 @import "../node_modules/react-select/less/default.less";
-@import (less) "../node_modules/webmaker-app-icons/css/ionicons.css";
-
-@font-face { font-family: "Ionicons"; src: url("vendor/webmaker-app-icons/fonts/ionicons.eot?v=2.0.1"); src: url("vendor/webmaker-app-icons/fonts/ionicons.eot?v=2.0.1#iefix") format("embedded-opentype"), url("vendor/webmaker-app-icons/fonts/ionicons.ttf?v=2.0.1") format("truetype"), url("vendor/webmaker-app-icons/fonts/ionicons.woff?v=2.0.1") format("woff"), url("vendor/webmaker-app-icons/fonts/ionicons.svg?v=2.0.1#Ionicons") format("svg"); font-weight: normal; font-style: normal; }
```
Comment (ScottDowne) [inline less/index.less]: Interesting, I thought we needed this for the fonts, but turns out we do not?

Comment (toolness) [inline less/index.less]: Yep see https://github.com/mozilla/teach.webmaker.org/issues/714#issuecomment-95921298 for more explanation.

Comment (ScottDowne) [inline less/index.less]: I think I see why it still works! Yay!

**Pull request merged**